data/base.go

```diff
@@ -0,0 +1,286 @@
+package data
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	valid "github.com/tidepool-org/platform/Godeps/_workspace/src/gopkg.in/bluesuncorp/validator.v8"
+	"github.com/tidepool-org/platform/Godeps/_workspace/src/labix.org/v2/mgo/bson"
+
+	"github.com/tidepool-org/platform/validate"
+)
+
+//used for all data types
+var validator = validate.NewPlatformValidator()
+
+func init() {
+	validator.RegisterValidation(timeStrTag, TimeStringValidator)
+	validator.RegisterValidation(timeObjTag, TimeObjectValidator)
+	validator.RegisterValidation(timeZoneOffsetTag, TimezoneOffsetValidator)
+	validator.RegisterValidation(payloadTag, PayloadValidator)
+	validator.RegisterValidation(annotationsTag, AnnotationsValidator)
+}
+
+//Base represents tha base types that all device data records contain
+type Base struct {
+	//required data
+	_ID      bson.ObjectId `bson:"_id" valid:"mongo,required"`
+	ID       string        `json:"id" bson:"id" valid:"required"`
+	UserID   string        `json:"userId" bson:"userId" valid:"required"`
+	DeviceID string        `json:"deviceId" bson:"deviceId" valid:"required"`
+	Time     string        `json:"time" bson:"time" valid:"timestr"`
+	Type     string        `json:"type" bson:"type" valid:"required"`
+	UploadID string        `json:"uploadId" bson:"uploadId" valid:"-"`
```

Comment:
We should have an uploadId validator as per my lengthy discussion with DuffLite.